draft-polli-ratelimit-headers.md

```diff
@@ -996,6 +997,12 @@ At this point you should stop increasing your request rate.
    Saturation conditions can be either dynamic or static: all this is out of
    the scope for the current document.
 
+   Newer HTTP versions like HTTP/2 can send multiple requests over the same
+   connection. A concurrency limit policy which drops connections may be problematic.
+
+   RateLimit headers enable sending *on the same connection* different limit values
+   on each response, depending on the policy scope (eg. per-user, per-custom-key, ..)
```

Comment:
I guess the fact connections come up often is related to people viewing this spec as a way to protect infrastructure rather than services. That is, "connections" is a detail of how infrastructure is accessed and is going to be defined in terms of the server and the network rather than the service being consumed. Other details could be brought up as well, such as the number of bytes the server could accept, which could also be limited.

OTOH, given how HTTP/2+ handles connections differently, adding information about them would probably make more sense than other infrastructure-related resources. However doing so would also bring in more questions, such as whether the limits would apply globally or not, for which period of time and which policy would be used, possibly replicating the information in this document for connections.

This risks entering the scoped rate limiting territory (and multiple rate limit information being returned at once), since in the end we are now protecting more than one resource in potentially different ways, and the combination might require lots of information being handed back to the client.

I think this document already provides value as is, and perhaps we could explain here that we are focusing on providing rate limiting information about a service-defined resource, which could vary dynamically as requests are performed, while leaving the door open for extension. Support for the use case of limiting connections could use other mechanisms, or be done later, if ever, via a more generic extension covering scoped and/or multiple rate limits.

Reply:
> the fact connections come up often is related to people viewing this spec as a way
> to protect infrastructure ...

It's meant to protect the client instead :D 

> Other details could be brought up as well

Absolutely whatever information that instead should not be disclosed too

> doing so would also bring in more questions, such as whether the limits
> would apply globally or not

Yep, though I'm not actually scared about that. The only reason we're not addressing `scope` *now* is because we want to do it in a way that is consistent with the rest of the HTTP specs, not because we won't do it at all.

> we are focusing on providing rate limiting information about a service-defined resource,
> which could vary dynamically as requests are performed

I thought that was quite clear throughout the document, but probably it's not.